docs/02_database/01_fields-tables-views/02_tables/01_tables-basics.md

```diff
@@ -100,38 +105,113 @@ tables {
 
-- Creating additional unique constraints to the table's primary key
-- Providing a performant record lookup (`unique` index)
-- Providing an index for a performant ranged lookup (`nonUnique` index, else `unique` index but searching on part of the key)
+### Enforcing uniqueness
+We have already noted that a unique index enforces uniqueness on that field. So you can add a unique index to a table to create a further constraint on the primary index.
 
-You can define indices on a table in a similar fashion to `primaryKey`. As with keys, indices are made up of one or more fields from the table. Where the indices are made up of multiple fields, the order of the fields matters if you want to search based on a partial index. This affects performance.
+Imagine a scenario where you have projects and users.
+
+- You have a PROJECTS table where each project has a unique PROJ_ID.
+- You have a USERS table where each project has a unique USER_ID.
+
+There is a many-to-many relationship between the two tables - projects can have more than one user, and users can belong to more than one project. To allocate users to projects, you need a third table: PROJECT_USERS.
+
+In the the PROJECT_USERS table, you only need two fields:
+
+- PROJ_ID
+- USER_ID
+
+Each record in this table states that PROJ_ID= x has USER_ID=y. But it is important that you do not duplicate any of these records - which could cause problems. So, in your PROJECT_USERS table, make PROJ_ID the primary index and add a unique index on PROJ_ID and USER_ID. The database will now prevent you from entering a record that has exactly the same PROJ_ID and USER_ID as an existing record.
+
+Here is our table definition for this:
 
 ```kotlin
-    indices {
-        unique {
-            SESSION_AUTH_TOKEN
-        }
-        nonUnique {
-            USER_NAME
-        }
-    }
+  table(name = "PROJECT_USERS", id = 11020) {
+      PROJ_ID
+      USER_ID
+      primaryKey {
+          PROJ_ID
+      }
+     indices {
+         unique {
+             PROJ_ID
+             USER_ID
+         }
+         nonUnique {
+             USER_ID
+         }
+     }
+  }
 ```
+You can clearly see the unique index on the two fields in the table above. As well as enforcing uniqueness, it also enables you to search for a record where USER_ID=y and USER_ID=y - so you can see if project x includes user y (and you can also make a partial search - see below).
+
+Note that the example also includes a non-unique index on USER_ID. This enables you to search for all records where USER_ID=y - so you can see exactly which projects the user belongs to.
+
+### Indices with multiple fields
+Any index you create - the primary key or other unique or non-unique indices - can have multiple fields.
+
+If you create an index with multiple fields, it is possible to make a search based on all the fields specified or to make a partial search based on some of the fields. However, there are strict limits to this.
+
```

Comment:
I would add here 

In the above example on the `PROJECT_USERS` table, this mean we can query on the primary key just on `PROJ_ID`, at which point it will show us the users associated with that project. We could also query on `PROJ_ID` and `USER_ID`. However, we cannot skip a field, therefore we need a separate index on `USER_ID`.